NEWS.md

```diff
@@ -96,6 +96,8 @@
 
 14. `setNumericRounding()` now invisibly returns the old rounding value instead of `NULL`, which is now consistent with similar behavior by `setwd()`, `options()`, etc. Thanks @MichaelChirico for the report and @joshhwuu for the fix.
 
+15. `dcast()` now issues a warning instead of a message when `fun.aggregate` is missing, defaulting to `length()`. This change makes potential issues more noticeable and easier to debug, [#5386](https://github.com/Rdatatable/data.table/issues/5386). The warning is classed as `dt_missing_agg_function_warning`, allowing for more targeted handling in user code. Thanks @MichaelChirico for the suggestion and @Nj221102 for the fix.
```

Comment:
We should be more specific here. "potential issues" is vague. As NEWS reader, my reaction is either (1) eyes glaze over & I miss something potentially important or (2) confusion as to whether I may be affected, and probably wind up spending too much time reading through details irrelevant to me --> frustration.

Better to offer enough details here so that the reader can quickly determine whether they need to look into this further for their own case.

The item is also _slightly_ incorrect -- the wording could be read to seem like we _always_ need to provide `fun.aggregate`.

I propose the following wording:

15. `dcast()` now issues a warning when `fun.aggregate` is used but not provided by the user. `fun.aggregate` defaults to `length` in this case; previously only a `message` was issued, but often relying on this default signals unexpected duplicates in the data, so a stricter class of signal was deemed more appropriate.

Reply:
done 👍